rs/types/types/src/crypto/canister_threshold_sig/idkg/proto_conversions.rs

```diff
@@ -237,6 +242,63 @@ impl TryFrom<&IDkgSignedDealingTupleProto> for SignedIDkgDealing {
     }
 }
 
+impl From<&EcdsaPreSignatureQuadruple> for EcdsaPreSignatureQuadrupleProto {
+    fn from(value: &EcdsaPreSignatureQuadruple) -> Self {
+        Self {
+            kappa_unmasked: Some(idkg_transcript_proto(&value.kappa_unmasked)),
+            lambda_masked: Some(idkg_transcript_proto(&value.lambda_masked)),
+            kappa_times_lambda: Some(idkg_transcript_proto(&value.kappa_times_lambda)),
+            key_times_lambda: Some(idkg_transcript_proto(&value.key_times_lambda)),
+        }
+    }
+}
+
+impl TryFrom<&EcdsaPreSignatureQuadrupleProto> for EcdsaPreSignatureQuadruple {
```

Comment:
Those are simple conversions, but it might still not harm to add some tests. Admittedly, there doesn't seem too much precedence in tests/proto_conversions.rs though.

Reply:
> but I'm getting some strange compile errors

This looks like a _quasi-circular dependency_ as Roman calls it: he explains this in detail in one of his blog posts [here](https://mmapped.blog/posts/03-rust-packages-crates-modules#quasi-circular). This could likely be fixed by turning the unit test into an integration test (by letting it live somewhere in `rs/types/types/test`).

However, if both conversion directions are exercises in some existing test (in `rs/replicated_state/src/metadata_state/tests.rs`) then this also seems sufficient.